### ISSUE TITLE
Comparing text with Unicode characters

### ISSUE THREAD

**Issue by MiattoRocha** [deitch/searchjs#51]
Hello people.

I'm designing a system and want to use SearchJS to my advanced search solution, but I'm having trouble with comparing strings with accents, since I'm working to a Latin company, our first users will be using the website in Portuguese. (é; ê; ã; á; â; à; õ; ó; ô; ç; etc)
Using the Text option as true, would be nice to have a accent folding.
I was looking for it online and found a option, javascript has the [String.prototype.toLocaleLowerCase()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/tolocalelowercase), using it instead could be a solution to i18n the SearchJS.

What do you think about this?

**Comment by deitch** [comment]
I like the idea, although we should leave it as an option. Some may _want_ to match only if the diacritics match.

> javascript has the String.prototype.toLocaleLowerCase(), using it instead could be a solution to i18n the SearchJS

I don't think that does it. My little bit of experience with it shows that it keeps the accents.

When you say "accent folding", you mean that, e.g. any of `àáâãäå` would match `a`, etc.?

**Comment by deitch** [comment]
There is a good sample [here](https://stackoverflow.com/questions/3686020/case-sensitive-accent-folding-in-javascript)  (for reference)

**Comment by MiattoRocha** [comment]
Yeah, I'm talking about match strings with same base character, exactly like you `àáâãäå` example.
I see, toLocaleLowerCase isn't a way to solve this.

That's is a good sample, but what did you think is best, implement a solution like that or use a lib to handle this replace?
Using a small but solid lib could avoid maintenance of adding a new letter every time someone need.

**Comment by deitch** [comment]
> Using a small but solid lib could avoid maintenance of adding a new letter every time someone need.

As long as we could package it in. This does work in the browser now, and I wouldn't want to lose that ability.

The sheer number of languages with diacritical marks would make it a maintenance challenge. I speak Hebrew fluently, and the same thing exists (fortunately, most people write it without the additional accent characters).

Do you know of any good libs to use?
